spec/amber/validations/params_spec.cr

```diff
@@ -12,8 +12,40 @@ module Amber
           required("last_name") { |v| v.str? & !v.empty? }
         end
 
-        validator.errors.size.should eq 0
         validator.valid?.should be_true
+        validator.errors.size.should eq 0
+      end
+
+      context "optional params" do
+        context "when missing" do
+          it "does not validate optional field" do
+            http_params = HTTP::Params.parse("&last_name=&middle=j")
+            validator = Validators::Params.new(http_params)
+
+            result = validator.validation do
+              optional(:name) { |v| v.empty? }
+              required("last_name") { |v| v.empty? }
+            end
+
+            validator.valid?.should be_false
+            validator.errors.size.should eq 1
+          end
+        end
+
+        context "when present" do
```

Comment:
maybe this is your positive test? 

either merge this with the context above or add another test in this context for when `validator.valid?.should be_true`